### ISSUE TITLE
[pyos][examples] Organization of notebook examples

### ISSUE THREAD

**Issue by sneakers-the-rat** [python-graphblas/python-graphblas#408]
I'll gather some thoughts on organization of the notebook examples that aren't bugs but style and readability suggestions here, rather than opening several issues:

## Global Structure

- It's not really clear to me if there is some intended order for these notebooks, and they seem to have mixed purpose: some basic syntax demos, some higher-level application examples, and what looks like a conference/workshop demo. I would suggest either structuring these by embedding them in the docs/linking to them from the docs with some explanation (links from the relevant sections of the API docs would also be nice, haven't looked to see if that exists yet). You might also consider subdirectories for different types of notebooks.

### Pagerank

- I am inferring the pagerank demo is intended as the introductory example from the "getting started" docs. It currently mixes some examples of basic syntax with the pagerank implementation itself, which is unclear - the example starts abruptly at "Compare with C" in [this version](https://github.com/python-graphblas/python-graphblas/blob/c3baea8d55df1ab015e27c9b9db16e40f64f446b/notebooks/Pagerank%20Demo.ipynb) without introduction, and the graph that was being used as an example in the first half is then changed to a binary graph. **recommendation:** split basic syntax into its own graph, either copy some of the description of the demo from the [docs](https://python-graphblas.readthedocs.io/en/stable/getting_started/primer.html#pagerank-using-linear-algebra) and link to it from that point in the docs (which would put it in two places, which might be less desirable), or, preferably, move the whole example + description to the notebook and embed in docs.
- Some narration and inline comments would help the pagerank demo a lot: eg. a larger graph is constructed with networkx and then converted to graphblas format - that's a nice feature, and highlights the relationship with networkx: that it can be used for some utility tasks that you can then move back and forth with graphblas. It would be worth commenting on that and potentially linking out to the other i/o functions in the API docs from there.

**pausing here for now, will edit any additional comments on example structure in here**

Related to: https://github.com/pyOpenSci/software-submission/issues/81

**Comment by eriknw** [comment]
Yeah, I agree these could use attention and would be nice to get some of them into the documentation.

Your assessment is correct: these are a hodgepodge of notebooks going back _years_ that have not been diligently kept up to date (except to run) or organized. I think it would be nice to update them to use current best practices (such as `gb.semiring.min_plus(A @ B)` instead of `A.mxm(B, gb.semiring.min_plus)`), and then figure out how to best integrate them into docs (and update the discussion in the notebook as appropriate).

Relatedly, we could even get more examples from here (most of these need cleaned up and updated even more):
https://github.com/metagraph-dev/grblas-recipes
